shell/components/auth/AllowedPrincipals.vue

```diff
@@ -56,7 +55,11 @@ export default {
 
   methods: {
     addPrincipal(id) {
-      addObject(this.authConfig.allowedPrincipalIds, id);
+      const currentIds = this.authConfig.allowedPrincipalIds;
+      // add new id and remove duplicates
+      const updatedIds = [...new Set([...currentIds, id])];
```

Comment:
I like it, It's elegant. Does Set preserve order?

We do use lodash though so I think we want to use https://lodash.com/docs/3.10.1#uniq. Notice they also have the unique() alias in case you'd want to use that.

Reply:
@codyrancher yes Set preserves order, but as you pointed out, I should've used lodash one, thanks!